### PR TITLE
WIP feat: Use manifest

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/google/model-transparency/issues/111

blocked on https://github.com/sigstore/sigstore-python/issues/628

### CODE REVIEW

**Comment by laurentsimon** [comment]
This is now working with https://github.com/sigstore/sigstore-python/pull/962

High-level:
clone sigstore-python
    make dev
    (source env/bin/activate)
    python3 -m pip install psutil==5.9.8
clone this repo
  run it as usual

**Comment by mihaimaruseac** [comment]
This is great! Thank you!

File: model_signing/_manifest.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            f = {
+                "path": p.path,
+                "digest": {
+                    str(p.hashed.algorithm): p.hashed.digest.hex(),
+                },
+            }
+            files += [f]
+        stmt = (
+            dsse._StatementBuilder()
+            .subjects(
```
Comment (susperius) [inline model_signing/_manifest.py:91]: Shouldn't the subjects be the files? 

My mental model had the statement subjects as the artifacts we want to sign so you get `subject[n].name = path` and `subject[n].digest = file_hash`. So we would only set the `predicate_type` field as `sigstore.dev/model-transparency/manifest/v1` and that describes everything we need to know about the subjects. Furthermore, the predicate could then hold metadata like `{"hash_algorithm": "abc"}`.

PLMK if there is something I don't get here.

See https://github.com/in-toto/attestation/blob/main/spec/v1/statement.md

Comment (laurentsimon) [reply model_signing/_manifest.py:91]: You're correct that's how intoto works, but there is some discussion here about whether we want to use intoto or not https://github.com/sigstore/model-transparency/issues/111

For this PR, I could not pack files in the subject due to some current limitation on the hash type accepted by sigstore-python library (they only accept known hashes, but w euse a parallel hash).

**Comment by mihaimaruseac** [comment]
Closing this one now that have #262 ... #267 and #276

**Pull request closed without merging**